clicking out of workspace file paths in terminal doesnt work if the file is already open

### Summary

e.g. if zed is open in `/home/user/dev/foo` clicking on a file path to `/home/user/dev/bar/a:10:5` does not take you to the file if `a` is already open 

```
2025-03-10T22:58:48.098078849-04:00 [INFO] open paths ["/home/user/dev/bar/a/"]
2025-03-10T22:58:48.098185513-04:00 [ERROR] Not a directory (os error 20)
```
found this in the logs after each subsequent time i try to click the file, seems to think its a directory for some reason?

### Zed Version and System Specs

Zed: v0.178.0 (Zed Dev d81e7683eacae1680da26b1e4bb20cd72391366c) 
OS: Linux X11 alpine 3.22.0_alpha20250108
Memory: 15.6 GiB
Architecture: x86_64
GPU: AMD Radeon RX 6950 XT (RADV NAVI21) || radv || Mesa 24.3.4

--- a/crates/terminal_view/src/terminal_view.rs
+++ b/crates/terminal_view/src/terminal_view.rs
@@ -1064,18 +1064,36 @@ fn possible_open_target(
 
     for worktree in &worktree_candidates {
         let worktree_root = worktree.read(cx).abs_path();
-        let paths_to_check = potential_paths
-            .iter()
-            .map(|path_with_position| PathWithPosition {
-                path: path_with_position
-                    .path
-                    .strip_prefix(&worktree_root)
-                    .unwrap_or(&path_with_position.path)
-                    .to_owned(),
-                row: path_with_position.row,
-                column: path_with_position.column,
-            })
-            .collect::<Vec<_>>();
+        let mut paths_to_check = Vec::with_capacity(potential_paths.len());
+
+        for path_with_position in &potential_paths {
+            if worktree_root.ends_with(&path_with_position.path) {
+                let root_path_with_posiition = PathWithPosition {
+                    path: worktree_root.to_path_buf(),
+                    row: path_with_position.row,
+                    column: path_with_position.column,
+                };
+                match worktree.read(cx).root_entry() {
+                    Some(root_entry) => {
+                        return Task::ready(Some(OpenTarget::Worktree(
+                            root_path_with_posiition,
+                            root_entry.clone(),
+                        )))
+                    }
+                    None => paths_to_check.push(root_path_with_posiition),
+                }
+            } else {
+                paths_to_check.push(PathWithPosition {
+                    path: path_with_position
+                        .path
+                        .strip_prefix(&worktree_root)
+                        .unwrap_or(&path_with_position.path)
+                        .to_owned(),
+                    row: path_with_position.row,
+                    column: path_with_position.column,
+                });
+            };
+        }
 
         let mut traversal = worktree
             .read(cx)

